src/main/java/hudson/scm/SubversionRepositoryStatus.java

```diff
@@ -161,17 +169,79 @@ void setJobProvider(JobProvider jobProvider) {
+                }
+                UUID remoteUUID = moduleLocation.getUUID(job, scm);
+                SVNURL repositoryRoot = moduleLocation.getRepositoryRoot(job, scm);
+                remoteUUIDCache.put(repositoryRoot.toString(), remoteUUID);
+                uuidAndRootPath = new SubversionRepoUUIDAndRootPath();
+                uuidAndRootPath.uuid = remoteUUID;
+                uuidAndRootPath.rootPath = repositoryRoot.getPath();
+            }
+            return uuidAndRootPath;
+        }
+
+        boolean doModuleLocationContainsAPathFromAffectedPath(String configuredRepoFullPath, String rootRepoPath, Set<String> affectedPath) {
+            boolean containsAnAffectedPath = false;
+
+            if( configuredRepoFullPath.startsWith(rootRepoPath) ) {
+                String remainingRepoPath = configuredRepoFullPath.substring(rootRepoPath.length());
+                if(remainingRepoPath.startsWith("/"))   remainingRepoPath=remainingRepoPath.substring(1);
+                String remainingRepoPathSlash = remainingRepoPath + '/';
+
+                for (String path : affectedPath) {
+                    if(path.equals(remainingRepoPath) /*for files*/ || 
+                            path.startsWith(remainingRepoPathSlash) /*for dirs*/ ||
+                            remainingRepoPath.length()==0 /*when someone is checking out the whole repo (that is, configuredRepoFullPath==rootRepoPath)*/) {
+                        // this project is possibly changed. poll now.
+                        // if any of the data we used was bogus, the trigger will not detect a change
+                        containsAnAffectedPath = true;
+                        break;
+                    }
+                }
+            }
+            return containsAnAffectedPath;
+        }
+
+        private void scheduleImediatePollingOfJob(Job job, SCMTrigger trigger, List<SvnInfo> infos) {
+            LOGGER.fine("Scheduling the immediate polling of "+job);
+
+            final RevisionParameterAction[] actions;
+            if (infos.isEmpty()) {
+                actions = new RevisionParameterAction[0];
+            } else {
+                actions = new RevisionParameterAction[] { new RevisionParameterAction(infos) };
+            }
+
+            trigger.run(actions);
+        }
+
         @Override
         public boolean onNotify(UUID uuid, long rev, Set<String> affectedPath) {
             boolean scmFound = false, triggerFound = false, uuidFound = false, pathFound = false;
-            Map<String, UUID> remoteUUIDCache = new HashMap<String, UUID>();
             LOGGER.fine("Starting subversion locations checks for all jobs");
-            for (Job p : this.jobProvider.getAllJobs()) {
-                SCMTriggerItem scmTriggerItem = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(p);
+            for (Job job : this.jobProvider.getAllJobs()) {
+                SCMTriggerItem scmTriggerItem = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(job);
```

Comment:
> Just to improve readability / maintainability

When you think existing source code really needs changes for readability, it is best to file that as a separate PR. (If the improvement is minor, I would sometimes reject the PR since the overhead added to anyone trying to merge/cherry-pick/blame nearby commits outweighs the advantage.)